vshard/router/init.lua

```diff
@@ -1741,7 +1756,13 @@ end
 local function legacy_cfg(cfg)
     if M.static_router then
         -- Reconfigure.
-        router_cfg(M.static_router, cfg, false)
+        router_cfg_fiber_safe(M.static_router, cfg, false)
+    elseif M.routers[STATIC_ROUTER_NAME] then
+        -- M.static_router is still not set but M.routers[STATIC_NAME] is
+        -- already there. This can only mean, that configuration is still in
+        -- progress. But simultaneous configuration is prohibited.
+        error(lerror.vshard(lerror.code.ROUTER_CFG_IS_IN_PROGRESS,
+                            STATIC_ROUTER_NAME))
```

Comment:
This one is for consistency of errors. Without that condition during the first configuration of static router via `vshard.router.cfg` in several fibers simultaneously we will get `ROUTER_ALREADY EXISTS` error, which is not very self-explaining as we allow `vshard.router.cfg` to be invoked several times for reconfiguration.